core/include/scipp/core/except.h

```diff
@@ -92,6 +92,10 @@ struct SCIPP_CORE_EXPORT CoordMismatchError : public std::runtime_error {
   using std::runtime_error::runtime_error;
 };
 
+struct SCIPP_CORE_EXPORT LabelMismatchError : public std::runtime_error {
+  using std::runtime_error::runtime_error;
+};
+
```

Comment:
Labels are just a special case of coords, I would not introduce a special exception for this, otherwise clients have to catch different types.